src/main/java/seedu/duke/parsermanager/ParseAddProperty.java

```diff
@@ -129,45 +131,51 @@ public ParseAddProperty(String addCommandDescription, PropertyList propertyList)
-            return new CommandAddProperty(propertyDetails);
-        } catch (EmptyDetailException e) {
-            throw new EmptyDetailException(MESSAGE_ADD_PROPERTY_WRONG_FORMAT);
-        } catch (MissingFlagException e) {
-            throw new MissingFlagException(MESSAGE_ADD_PROPERTY_WRONG_FORMAT);
-        } catch (IncorrectFlagOrderException e) {
-            throw new IncorrectFlagOrderException(MESSAGE_ADD_PROPERTY_WRONG_FORMAT);
+    public Command parseCommand() throws ParseAddPropertyException {
+        checkForEmptyAddPropertyDetails(commandDescription);
+        ArrayList<String> propertyDetails = processCommandAddPropertyDetails(commandDescription);
+        validatePropertyDetails(propertyDetails);
+        return new CommandAddProperty(propertyDetails);
+    }
+
+    private void checkForEmptyAddPropertyDetails(String commandDetail) throws EmptyAddPropertyDetailException {
+        boolean isEmptyDetail = isEmptyString(commandDetail);
+        if (isEmptyDetail) {
+            throw new EmptyAddPropertyDetailException();
         }
     }
 
     private ArrayList<String> processCommandAddPropertyDetails(String rawCommandDetail)
-            throws MissingFlagException, IncorrectFlagOrderException {
+            throws MissingAddPropertyFlagException, IncorrectAddPropertyFlagOrderException {
         int[] flagIndexPositions = getFlagIndexPositions(rawCommandDetail, ADD_PROPERTY_FLAGS);
         checkForMissingPropertyFlags(flagIndexPositions);
         checkPropertyFlagsOrder(flagIndexPositions);
         return extractPropertyDetails(rawCommandDetail, flagIndexPositions);
     }
 
-    private void checkForMissingPropertyFlags(int[] flagIndexPositions) throws MissingFlagException {
+    private void checkForMissingPropertyFlags(int[] flagIndexPositions) throws MissingAddPropertyFlagException {
         for (int flagIndex : flagIndexPositions) {
-            if (!checkForFlagPresence(flagIndex)) {
-                throw new MissingFlagException(EXCEPTION);
+            boolean hasPropertyFlag = checkForFlagPresence(flagIndex);
+            if (!hasPropertyFlag) {
+                throw new MissingAddPropertyFlagException();
             }
         }
     }
 
-    private void checkPropertyFlagsOrder(int[] flagIndexPositions) throws IncorrectFlagOrderException {
+    private void checkPropertyFlagsOrder(int[] flagIndexPositions) throws IncorrectAddPropertyFlagOrderException {
         for (int flagIndex = 0; flagIndex < PROPERTY_FLAG_SIZE - UNIT_VALUE; flagIndex++) {
             checkForCorrectFlagOrder(flagIndexPositions[flagIndex], flagIndexPositions[flagIndex + UNIT_VALUE]);
         }
     }
 
+    private void checkForCorrectFlagOrder(int flagPosition, int nextFlagPosition) throws
+            IncorrectAddPropertyFlagOrderException {
+        boolean hasCorrectOrder = (flagPosition < nextFlagPosition);
+        if (!hasCorrectOrder) {
+            throw new IncorrectAddPropertyFlagOrderException();
+        }
```

Comment:
Good exception throwing here. I really like how your exception is really helpful in avoiding runtime errors.